packages/gatsby-plugin-netlify-cms/src/gatsby-node.js

```diff
@@ -145,6 +149,30 @@ exports.onCreateWebpackConfig = (
       devtool: stage === `develop` ? `cheap-module-source-map` : `source-map`,
     }
 
+    config.module.rules.push({
+      test: /gatsby\/cache-dir.*\.js$/,
+      loader: require.resolve(`babel-loader`),
+      options: {
+        presets: [
+          require.resolve(`@babel/preset-react`),
+          [
+            require.resolve(`@babel/preset-env`),
+            {
+              shippedProposals: true,
+              useBuiltIns: `entry`,
+            },
+          ],
+        ],
+        plugins: [require.resolve(`@babel/plugin-proposal-class-properties`)],
+      },
+    })
+
+    // Transpile Gatsby module because Gatsby includes un-transpiled ES6 code.
+    config.module.rules.exclude = [/node_modules\/(?!(gatsby)\/)/]
+
+    // Prefer Gatsby ES6 entrypoint (module) over commonjs (main) entrypoint
+    config.resolve.mainFields = [`browser`, `module`, `main`]
```

Comment:
if not mistaken it's the default https://webpack.js.org/configuration/resolve/#resolvemainfields

Reply:
Good point, removing. This setting was directly taken from: https://www.gatsbyjs.org/docs/visual-testing-with-storybook/ Given that storybook runs on webpack 4 and the webpack docs is referencing v4, not exactly sure why this setting is relevant.